src/container.cpp

```diff
@@ -199,7 +199,7 @@ void Container::onUpdateContainerItem(uint32_t index, Item* oldItem, Item* newIt
 
 	//send to client
 	for (Creature* spectator : spectators) {
-		spectator->getPlayer()->sendUpdateContainerItem(this, index, newItem);
```

Comment:
Wait, why do we have different types for `index`? It should be either both `uint32_t` or both `uint16_t` instead of casting. This may be hiding an issue.